Api/Controllers/LevelsController.cs

```diff
@@ -1,35 +1,59 @@
 ﻿using DomainLayer.Constants;
+using DomainLayer.Entities;
 using Microsoft.AspNetCore.Mvc;
+using RepositoryLayer.Repositories;
+using ServiceLayer.Business;
+using WebApiLayer.UserFeatures.Requests;
 
 namespace WebApiLayer.Controllers;
 
 [Route(Constants.HTTP.API_VERSION + "/gms/levels")]
 public class LevelsController : BaseController
 {
+    private readonly ILevelServices _levelServices;
+    private readonly IGenericRepository<LevelEntity> _levelRepo;
+    public LevelsController(ILevelServices levelServices, IGenericRepository<LevelEntity> levelRepo)
+    {
+        _levelServices = levelServices;
+        _levelRepo = levelRepo;
+    }
+
     [HttpGet]
-    public IEnumerable<string> Get()
+    public async Task<IActionResult> GetLevels()
     {
-        return new string[] { "value1", "value2" };
+        var levels = await _levelServices.List();
+        return Ok(levels);
     }
 
     [HttpGet("{id}")]
-    public string Get(int id)
+    public async Task<IActionResult> GetLevel(Guid id)
     {
-        return "value";
+        var level = await _levelServices.GetById(id);
+        return Ok(level);
     }
 
     [HttpPost]
-    public void Post([FromBody] string value)
+    public async Task<IActionResult> PostLevel([FromBody] CreateLevelsController level)
```

Comment:
CreateLevel is more correct for this method, name method corresponding with Get, Create, Update, Delete